### PR TITLE
feat(space): add fill support

### PR DESCRIPTION
**Pull request by adaex**
Add fill support for space

re #2675

* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by adaex** [comment]
@JeremyWuuuuu  I'm not sure whether my coding style matches this repo. Please let me know if I have any mistakes.

**Comment by JeremyWuuuuu** [comment]
> @JeremyWuuuuu I'm not sure whether my coding style matches this repo. Please let me know if I have any mistakes. 🙇🙇🙇

NP, I will guide you through this.

**Comment by JeremyWuuuuu** [comment]
@adaex Can we use something like this?
https://css-tricks.com/filling-space-last-row-flexbox/

**Comment by JeremyWuuuuu** [comment]
I think adding width to `100%` might not be the best solution here.

**Comment by adaex** [comment]
> @adaex Can we use something like this?
> https://css-tricks.com/filling-space-last-row-flexbox/

Great, I think this is a particularly good solution. I will try to do it and test.

**Comment by adaex** [comment]
@JeremyWuuuuu 

I have completed it initially, and the effect is like this:

sapce-fill-demo


But I am not sure if I need to add another parameter `fillPercentage`, this parameter can customize the filling ratio, the effect is as follows:

sapce-fill-percentage-demo


I don't know whether this feature is necessary or how to name this parameter.

File: packages/space/src/useSpace.ts [hunk 1]
```diff
@@ -50,6 +50,19 @@ export const defaultProps = {
...
+    type: Boolean,
+    default: false,
+  },
+
+  fillPercentage: {
+    type: Number,
+    default: 100,
+    validator: (val: unknown) => {
+      return isNumber(val) && val > 0 && val <= 100
+    },
```
Comment (JeremyWuuuuu) [inline packages/space/src/useSpace.ts:63]: I don't think the validator is necessary right here though. Maybe we change change the name from <s> fillPercentage</s> to  `fillRatio`

**Comment by JeremyWuuuuu** [comment]
Also, can you add a documentation item which shows how to use `fill`?

File: packages/space/src/useSpace.ts (hunk 1 above)
Comment (adaex) [reply packages/space/src/useSpace.ts]: I quite agree with you. `fillRatio` is much better than `fillPercentage`.

**Comment by adaex** [comment]
> Also, can you add a documentation item which shows how to use `fill`?

OK, I'm done.

**Comment by JeremyWuuuuu** [comment]
@adaex You did a really great job! ElementPlus is lucky to have you as a contributor, the code LGTM, I am happy to approve as is.

**Review by jw-foss — APPROVED** [review]

**Comment by adaex** [comment]
> @adaex You did a really great job! ElementPlus is lucky to have you as a contributor, the code LGTM, I am happy to approve as is.

Thank you. 🙇🙇🙇

**Pull request merged**